File: tests/robotests/src/com/android/settings/ResetNetworkConfirmTest.java
/*
 * Copyright (C) 2018 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.android.settings;

import static com.google.common.truth.Truth.assertThat;

import static org.mockito.Mockito.spy;

import android.view.LayoutInflater;
import android.widget.TextView;

import androidx.fragment.app.FragmentActivity;

import com.android.settings.testutils.shadow.ShadowBluetoothAdapter;
import com.android.settings.testutils.shadow.ShadowRecoverySystem;

import org.junit.After;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnit;
import org.mockito.junit.MockitoRule;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.android.util.concurrent.PausedExecutorService;
import org.robolectric.annotation.Config;
import org.robolectric.shadows.ShadowLooper;
import org.robolectric.shadows.ShadowPausedAsyncTask;

@RunWith(RobolectricTestRunner.class)
@Config(shadows = {ShadowRecoverySystem.class, ShadowBluetoothAdapter.class})
public class ResetNetworkConfirmTest {
    @Rule
    public final MockitoRule mMockitoRule = MockitoJUnit.rule();

    private static final String TEST_PACKAGE = "com.android.settings";

    private FragmentActivity mActivity;

    @Mock
    private ResetNetworkConfirm mResetNetworkConfirm;
    private PausedExecutorService mExecutorService;

    @Before
    public void setUp() {
        mExecutorService = new PausedExecutorService();
        ShadowPausedAsyncTask.overrideExecutor(mExecutorService);
        mResetNetworkConfirm = new ResetNetworkConfirm();
        mActivity = spy(Robolectric.setupActivity(FragmentActivity.class));
        mResetNetworkConfirm.mActivity = mActivity;
    }

    @After
    public void tearDown() {
        ShadowRecoverySystem.reset();
    }

    @Test
    public void testResetNetworkData_notResetEsim() {
        mResetNetworkConfirm.mResetNetworkRequest =
                new ResetNetworkRequest(ResetNetworkRequest.RESET_NONE);
        mResetNetworkConfirm.mResetSubscriptionContract =
                new ResetSubscriptionContract(mActivity,
                mResetNetworkConfirm.mResetNetworkRequest) {
            @Override
            public void onSubscriptionInactive(int subscriptionId) {
                mActivity.onBackPressed();
            }
        };

        mResetNetworkConfirm.mFinalClickListener.onClick(null /* View */);
        mExecutorService.runAll();
        ShadowLooper.idleMainLooper();

        assertThat(ShadowRecoverySystem.getWipeEuiccCalledCount()).isEqualTo(0);
    }

    @Test
    public void setSubtitle_eraseEsim() {
        mResetNetworkConfirm.mResetNetworkRequest =
                new ResetNetworkRequest(ResetNetworkRequest.RESET_NONE);
        mResetNetworkConfirm.mResetNetworkRequest.setResetEsim(TEST_PACKAGE);

        mResetNetworkConfirm.mContentView =
                LayoutInflater.from(mActivity).inflate(R.layout.reset_network_confirm, null);

        mResetNetworkConfirm.setSubtitle();

        assertThat(((TextView) mResetNetworkConfirm.mContentView
                .findViewById(R.id.reset_network_confirm)).getText())
                .isEqualTo(mActivity.getString(R.string.reset_network_final_desc_esim));
    }

    @Test
    public void setSubtitle_notEraseEsim() {
        mResetNetworkConfirm.mResetNetworkRequest =
                new ResetNetworkRequest(ResetNetworkRequest.RESET_NONE);

        mResetNetworkConfirm.mContentView =
                LayoutInflater.from(mActivity).inflate(R.layout.reset_network_confirm, null);

        mResetNetworkConfirm.setSubtitle();

        assertThat(((TextView) mResetNetworkConfirm.mContentView
                .findViewById(R.id.reset_network_confirm)).getText())
                .isEqualTo(mActivity.getString(R.string.reset_network_final_desc));
    }
}

File: tests/spa_unit/src/com/android/settings/system/reset/ResetNetworkConfirmTest.kt
/*
 * Copyright (C) 2024 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.android.settings.system.reset

import android.content.Context
import android.view.LayoutInflater
import android.widget.TextView
import androidx.fragment.app.testing.launchFragment
import androidx.test.core.app.ApplicationProvider
import androidx.test.ext.junit.runners.AndroidJUnit4
import com.android.settings.R
import com.android.settings.ResetNetworkRequest
import com.google.common.truth.Truth.assertThat
import kotlinx.coroutines.runBlocking
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.kotlin.any
import org.mockito.kotlin.never
import org.mockito.kotlin.spy
import org.mockito.kotlin.verify

@RunWith(AndroidJUnit4::class)
class ResetNetworkConfirmTest {
    private val context: Context = spy(ApplicationProvider.getApplicationContext()) {}

    private val scenario = launchFragment<ResetNetworkConfirm>()

    @Test
    fun resetNetworkData_notResetEsim() {
        scenario.recreate().onFragment { fragment ->
            fragment.resetNetworkRequest = ResetNetworkRequest(ResetNetworkRequest.RESET_NONE)

            runBlocking { fragment.onResetClicked() }

            verify(context, never()).getSystemService(any())
        }
    }

    @Test
    fun setSubtitle_eraseEsim() {
        scenario.onFragment { fragment ->
            fragment.resetNetworkRequest =
                ResetNetworkRequest(ResetNetworkRequest.RESET_NONE).apply {
                    setResetEsim(context.packageName)
                }

            val view = fragment.onCreateView(LayoutInflater.from(context), null, null)

            assertThat(view.requireViewById<TextView>(R.id.reset_network_confirm).text)
                .isEqualTo(context.getString(R.string.reset_network_final_desc_esim))
        }
    }

    @Test
    fun setSubtitle_notEraseEsim() {
        scenario.onFragment { fragment ->
            fragment.resetNetworkRequest = ResetNetworkRequest(ResetNetworkRequest.RESET_NONE)

            val view = fragment.onCreateView(LayoutInflater.from(context), null, null)

            assertThat(view.requireViewById<TextView>(R.id.reset_network_confirm).text)
                .isEqualTo(context.getString(R.string.reset_network_final_desc))
        }
    }
}
